src/librustc_span/source_map.rs

```diff
@@ -141,27 +141,29 @@ pub struct SourceMap {
     // This is used to apply the file path remapping as specified via
     // `--remap-path-prefix` to all `SourceFile`s allocated within this `SourceMap`.
     path_mapping: FilePathMapping,
+    hash_kind: SourceFileHashAlgorithm,
```

Comment:
Could use a comment, but the main thing I want to say is that I'm glad this no longer needs to be mutable!